StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs

```diff
@@ -507,12 +507,12 @@ void MethodName()
     {
         do
         {
-            Bar();
+            MethodName();
         }
         while (false);
 ";
 
-            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode, allowUnsupportedDiagnostics: true).ConfigureAwait(false);
```

Comment:
This test contains invalid C# syntax. The closing brace for the method is not present, resulting in a compiler being produced for it in the verify case.
As I wanted to keep the `Assert` in the `CodeFixVerifier` signalling that a test case is producing unexpected errors, I had to go to quite a bit of effort to make this test case work.